galley/pkg/runtime/state_test.go

```diff
@@ -309,6 +300,10 @@ func TestState_String(t *testing.T) {
 }
 
 func newTestState() *State {
-	return newState(groups.Default, resources.TestSchema, cfg, publish.NewStrategyWithDefaults(),
-		snapshot.New(groups.IndexFunction))
+	stateStrategy := publish.NewStrategyWithDefaults()
+	stateListener := processing.ListenerFromFn(func(c resource.Collection) {
+		// When the state indicates a change occurred, update the publishing strategy
+		stateStrategy.OnChange()
+	})
+	return newState(resources.TestSchema, cfg, stateListener)
```

Comment:
At this point, you can simply pass the stateStrategy to newState, and avoid indirection

Reply:
Actually this is the way the code was originally, before some of my previous refactorings ... the processor owned the strategy. I think I'd rather just keep it this way at this point.

Reply:
I am suggesting getting rid of the Listener abstraction. It doesn't seem to be serving a useful purpose here.

Reply:
well, it does decouple the state and processor so that they don't have to share private fields. So logically ... sure, it doesn't change anything, but it is an slight improvement for code health and it also introduces a coming-soon feature for your dataflow model.